AutoSteamApp/Process Memory/SteamworksData.cs

```diff
@@ -0,0 +1,235 @@
+﻿using AutoSteamApp.Helpers;
+using GregsStack.InputSimulatorStandard.Native;
+using Logging;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Windows.Forms;
+
+namespace AutoSteamApp.ProcessMemory
+{
+
+    public class SteamworksData
+    {
+
+        #region Fields
+
+        /// <summary>
+        /// The Monster Hunter World: Iceborne process to load player data from.
+        /// </summary>
+        Process MHWProcess;
+
+        /// <summary>
+        /// The address which begins the "steamworks" object.
+        /// </summary>
+        ulong SteamworksAddress;
+
+        /// <summary>
+        /// Address which holds the current expected sequence of the game.
+        /// </summary>
+        ulong SequenceAddress;
+
+        /// <summary>
+        /// Address which indicates whether the button input was registered.
+        /// </summary>
+        ulong ButtonPressedCheckAddress;
+
+        /// <summary>
+        /// Address used to read what phase the steamworks is in. 
+        /// </summary>
+        ulong PhaseAddress;
+
+        /// <summary>
+        /// Address used to read what sub-phase the steamworks is in. When in the bonus or settled phase.
+        /// </summary>
+        ulong SecondPhaseAddress;
+
+        /// <summary>
+        /// Address used to read the rarity of the reward.
+        /// </summary>
+        ulong RarityAddress;
+
+        /// <summary>
+        /// Address used to read the current steam gauge progress.
+        /// </summary>
+        ulong SteamGaugeAddress;
+        
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// The value used to check whether the game has registered the input press.
+        /// </summary>
+        public byte InputPressStateCheck
```

Comment:
property can be simplified `public byte InputPressStateCheck => MemoryHelper.Read<byte>(MHWProcess, ButtonPressedCheckAddress);`

Reply:
If you want to use Expression-bodied members that's fine by me. I think it is antithesis to c#, however I am more than happy to do it your preferred way.